src/Service/PromotionsService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository;
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Domain\Exception\DataNotFetchedException;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\PromotionMapper;
+use DateTimeImmutable;
+
+class PromotionsService extends AbstractService
+{
+    /** @var PromotionRepository */
+    protected $repository;
+
+    public function __construct(
+        PromotionRepository $repository,
+        PromotionMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    /**
+     * @return Promotion[]
+     * @throws DataNotFetchedException if we get a promotion but there is nothing being
```

Comment:
This method does not directly throw an exception. `@throws` should only be placed functions that directly throw, and not on functions that call functions that throw. Otherwise basically every function shall end up decorated with an `@throws` which would be very noisy.